## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [1]:
import pandas as pd
import numpy as np

In [2]:
data_path = '../Статистика/data.csv'
df = pd.read_csv(data_path)
df

Unnamed: 0       data
0            0  17.499453
1            1  19.662399
2            2   7.182823
3            3  29.841625
4            4   9.239386
..         ...        ...
95          95  26.244944
96          96  22.282832
97          97  18.697188
98          98  27.041189
99          99   5.938381

[100 rows x 2 columns]

In [3]:
data = np.array(df.data)
data

array([17.49945292, 19.66239904,  7.18282343, 29.84162485,  9.23938649,
       14.94951581, 23.0172885 , 12.52827148, 13.65228669, 14.54595431,
       23.30872427, 33.75324808, 20.24923636, 13.29244733, 23.23434992,
       16.4230418 , 19.88521702, 27.05000732, 15.5127743 , 20.05415151,
       14.73135264, 19.06139498, 21.53942271, 14.06732571, 17.9670682 ,
       18.58289581, 16.17406993, 12.87432628, 11.47269664, 19.07902883,
       18.38565824, 33.38820073,  5.39139454, 20.67635903, 22.22266722,
       28.15780318, 23.01114324, 14.93471778, 20.00005857, 23.25411543,
       18.11895082, 24.62607043,  8.79145607, 30.387108  , 28.80606806,
       17.98593597, 23.66804468, 20.28782355, 15.02518827, 20.52626131,
       26.00219532, 17.71344489, 17.74598346, 19.55317542, 22.60097798,
       27.67027538, 16.19192417, 23.05037746, 21.29669604,  8.84832568,
       17.48410111, 19.20602661, 19.76257856, 21.9560206 ,  7.75806171,
       20.27753314, 15.93394654, 11.36336584, 23.14577858, 24.41

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [4]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [5]:
from statsmodels.stats.weightstats import ztest
import math

In [6]:
null_mean = 20
alpha = 0.05

In [7]:
np.mean(data), np.std(data)

(19.37755319670441, 6.222214863419309)

In [8]:
# Оскільки розмір вибірки більше 30, то обраний z-test
ztest_Score, p_value = ztest(data, value = null_mean, alternative='larger')
ztest_Score, p_value

(-0.9953477389335053, 0.8402164165464012)

In [9]:
if(p_value < alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


Висновок: Ні, ми не маємо зараз статистично значущу різницю у розмірі середнього чеку. Статистично значущо підвищити середній чек за останній місяць не вдалося.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [10]:
data_path2 = '../Статистика/data_2.csv'
df2 = pd.read_csv(data_path2)
df2

data
0     5.940159
1    29.189344
2    11.383502
3    15.434098
4    19.896752
..         ...
115  31.304916
116  15.822676
117  19.291394
118  19.036747
119  18.847899

[120 rows x 1 columns]

In [11]:
data2 = np.array(df2.data)
data2

array([ 5.94015879, 29.18934391, 11.38350244, 15.43409823, 19.89675209,
       26.35462741, 21.71541042, 30.07928738, 18.559706  ,  8.98262482,
       21.75095478, 21.99252789, 16.37737276, 18.36827303, 15.96373794,
       30.98838431, 16.13256609, 11.01409438, 21.96585507,  9.46690988,
       23.42563685, 21.53704461, 13.26051457, 17.14026368, 15.42554536,
       30.6793921 , 21.26136766, 12.18702858, 10.69691295, 21.57137244,
       19.44915472, 15.29409103, 21.20725805, 18.5615031 , 27.27465429,
       22.71259245, 19.41187696, 21.98647106, 13.33332309, 26.42199496,
       23.87611696, 16.09133552, 20.95972522, 15.47445472, 32.79323825,
       16.42470817, 23.61838166, 22.92533987, 14.82328184, 19.14936908,
       17.61708292, 20.76144078, 20.88864519, 22.17749704, 24.44099555,
       17.42064342, 19.01311241, 17.11869915, 14.66455092,  9.59578906,
       23.93038503, 18.80003123, 16.62880551,  9.30221669, 10.74561613,
        5.00093505, 18.64262674, 23.06962246, 23.27815123, 19.56

In [12]:
data2.shape

(120,)

In [13]:
np.mean(data), np.mean(data2)

(19.37755319670441, 19.37322276512519)

In [14]:
np.std(data), np.std(data2)

(6.222214863419309, 5.936369567972301)

In [15]:
ztest(data, data2, value=0)

(0.005246673456808504, 0.9958137794593356)

Висновок: Статистика для двох вибірок z-критерію становить 0.00525, а відповідне p-value дорівнює 0.9958.

Оскільки p-value > 0.05, то нульова гіпотеза не може бути відхилена. Тобто середній рівень продажів за два порівнювальні місяці значимо не відрізняється. Це підтверджується і значеннями середніх, що були виведені вище і збігаються до 2-го знаку після коми.